cmd/anonymizer/app/anonymizer/anonymizer_test.go

```diff
@@ -73,6 +75,64 @@ var span2 = &model.Span{
 	StartTime: time.Unix(300, 0),
 }
 
+func TestNew(t *testing.T) {
+	nopLogger := zap.NewNop()
+	tempDir := t.TempDir()
+
+	file, err := os.CreateTemp(tempDir, "mapping.json")
+	assert.NoError(t, err)
+
+	_, err = file.Write([]byte(`
+{
+    "services": {
+		"api": "hashed_api"
+	},
+	"operations": {
+		"[api]:delete": "hashed_api_delete"
+	}
+}
+`))
+	assert.NoError(t, err)
+
+	anonymizer := New(file.Name(), Options{}, nopLogger)
+	assert.NotNil(t, anonymizer)
+}
+
+func TestAnonymizer_SaveMapping(t *testing.T) {
+	nopLogger := zap.NewNop()
+	mapping := mapping{
+		Services: make(map[string]string),
+		Operations: make(map[string]string),
+	}
+
+	tests := []struct{
+		name string
+		mappingFile string
+	}{
+		{
+			name:        "fail to write mapping file",
+			mappingFile: "",
+		},
+		{
+			name:        "save mapping file successfully",
+			mappingFile: filepath.Join(t.TempDir(), "mapping.json"),
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
```

Comment:
nit: this will work but don't think it's necessary; it's useful when referring to its pointer reference or if being used concurrently.